main/influxdb_watcher.go

```diff
@@ -136,6 +136,32 @@ func queryDB(broker *InfluxdbBroker, cmd string, ch chan []client.Result, cherr
 	ch <- response.Results
 }
 
+func (broker *InfluxdbBroker) checkDatabaseExists(timeout time.Duration) bool {
+	cmd := "SHOW DATABASES"
+	ch := make(chan []client.Result)
+	cherr := make(chan error)
+	go queryDB(broker, cmd, ch, cherr)
+	for {
```

Comment:
What is the point of this for? Everything in the select either returns true or false so it will never loop.......

Reply:
Honestly, the only examples I saw of a select statement had a for loop around it. That is my bad to assume.